pkg/streamingpromql/query.go

```diff
@@ -493,7 +493,10 @@ func (q *Query) Exec(ctx context.Context) *promql.Result {
 		return &promql.Result{Err: compat.NewNotSupportedError(fmt.Sprintf("unsupported result type %s", parser.DocumentedType(q.statement.Expr.Type())))}
 	}
 
-	q.result.Warnings = *q.annotations
+	// To make comparing to Prometheus' engine easier, only return Annotations if there are some, otherwise, return nil.
```

Comment:
[nit] 

```suggestion
	// To make comparing to Prometheus' engine easier, only return the annotations if there are some, otherwise, return nil.
```